### ISSUE TITLE
E tensorflow/stream_executor/cuda/cuda_driver.cc:406] failed call to cuInit: CUDA_ERROR_NO_DEVICE

### ISSUE THREAD

**Issue by jkzhang7** [tensorflow/tensorflow#40920]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): pip install tensorflow-gpu==1.8.0
- TensorFlow version (use command below): v1.8.0.0-g93bc2e2072
- Python version: Python 2.7
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: CUDA 9.0 / cuDNN 7.5.1
- GPU model and memory: 

**Describe the current behavior**
When the training start, the error is shown as following:
> 2020-06-29 14:30:40.010427: E tensorflow/stream_executor/cuda/cuda_driver.cc:406] failed call to cuInit: CUDA_ERROR_NO_DEVICE

And when I am trying use 'nvidia-smi' command, the error is shown as following:
> NVIDIA-SMI has failed because it couldn't communicate with the NVIDIA driver. Make sure that the latest NVIDIA driver is installed and running.

I tried the solution descried in https://github.com/tensorflow/tensorflow/issues/16860, and I tried reinstall tensorflow-gpu, They all did not work... Does anyone know how to deal with it? Really appreciate it!

**Comment by amahendrakar** [comment]
@jkzhang7,
Could you please provide the complete code/commands you have run before running into this issue.

Also, TensorFlow 1.8 is outdated could you please install TensorFlow 1.15.3 or 2.2 using [this](https://www.tensorflow.org/install/gpu#linux_setup) official guide. Thanks!

**Comment by jkzhang7** [comment]
@amahendrakar 
I used the customized code here. For some reason, the TensorFlow 1.15.3 is not compatible with my code. Also, when I used the following command, 

`from tensorflow.python.client import device_lib` 
`device_lib.list_local_devices()` 

The information is shown as following,

![image](https://user-images.githubusercontent.com/25946506/86301617-2a3ada80-bbbb-11ea-870e-dfc8b1c1b617.png)

Does it mean there is some mess with my GPU driver?

Thanks!
